modules/internal/Atomics.chpl

```diff
@@ -467,15 +467,15 @@ module Atomics {
       return ret;
     }
     inline proc add(value:uint(8), order:memory_order = memory_order_seq_cst):void {
-      on this do atomic_fetch_add_explicit_uint_least8_t(_v, value, order);
+      on this do local atomic_fetch_add_explicit_uint_least8_t(_v, value, order);
```

Comment:
It appears that this pass detects whether or not a primitive is inside a local block, and can consider a primitive 'fast' even if wide pointers are involved.

Do you see performance improvements from this?

Something that may help would be to teach the compiler that 'this' in 'on this do' is local within the on-statement. That should avoid related wide pointers for accessing fields like '_v'. This optimization has been on my list for a while, but I haven't found a motivating example for improvement. Maybe this is it?